File: 0525-contiguous-array/0525-contiguous-array.cs
public class Solution {
    public int FindMaxLength(int[] nums) {
        if(nums.Length == 1)
        {
            return 0;
        }
        int max = 0;
        int sum = 0;
        Dictionary<int,int> map=new();
        for(int i=0;i<nums.Length;i++)
        {
            if(nums[i]==0)
            {
                sum--;
            }
            else
            {
                sum++;
            }
            if(!map.ContainsKey(sum) && sum != 0)
            {
                 map.Add(sum,i);
            }
            else if(map.ContainsKey(sum))
            {
                max = Math.Max(max,i-map[sum]);
            }
            else if(sum==0)
            {
                max = Math.Max(max,i+1);
            }
        }
        return max;
    }
}

File: 0525-contiguous-array/0525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        map.put(0, -1);
        int sum = 0;
        int ans = 0;
        int idx = 0;
        for(int num : nums)
        {
            if(num == 0)
            {
                sum--;
            }
            else
            {
                sum++;
            }
            if(sum == 0)
            {
                ans = Math.max(ans, idx-map.get(0));
            }
            if(!map.containsKey(sum))
            {
                map.put(sum, idx);
            }
            else
            {
                ans = Math.max(ans, idx - map.get(sum));
            }
            idx++;
        }
        return ans;
    }
}